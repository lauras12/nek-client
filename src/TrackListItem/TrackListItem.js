import React from 'react';
import TokenService from '../services/token-service';
import config from '../config';
import { withRouter } from 'react-router-dom';
import './TrackListItem.css';
import { animateScroll as scroll } from 'react-scroll';

function TrackListItem(props) {

    const handleClick = () => {
        const token = TokenService.hasAuthToken(config.TOKEN_KEY);
        scroll.scrollToTop();
        if (!token) {
            props.history.push(`/login`);
        }
        else if (props.location.pathname === '/') {
            props.history.push('/hikeForm');
        }
        else {
            props.history.push(`/hike/${props.id}`);
        }
    }

    return (
        <div className='item' >
            <li className='li' onClick={() => handleClick()}>
                <h2>{props.name}</h2>
                <h2>{props.sanskrit}</h2>
                <img src={props.img} alt='track, asana'/>
            </li>
        </div>
    );
}

export default withRouter(TrackListItem);