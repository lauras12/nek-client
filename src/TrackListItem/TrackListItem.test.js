import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import TrackListItem from './TrackListItem';
import {BrowserRouter as BR} from 'react-router-dom';

describe('TrackListItem component', () => {
    const props = {
        name: 'chair',
        sanskrit: 'utkatasana',
        img: '',
        location: {pathname: '/'},
        history: {push : '/'}
    };

    it('renders without crashing', () => {
         const div = document.createElement('div');
         ReactDom.render(<BR><TrackListItem {...props} /></BR>, div);
         ReactDom.unmountComponentAtNode(div);
    });

    it('renders UI as expected', () => {
        const item = renderer.create(<BR><TrackListItem {...props}/></BR>)
        expect(item.toJSON()).toMatchSnapshot();
    });
});
