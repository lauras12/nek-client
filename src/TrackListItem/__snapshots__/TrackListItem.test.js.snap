// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackListItem component renders UI as expected 1`] = `
<div
  className="item"
>
  <li
    className="li"
    onClick={[Function]}
  >
    <h2>
      Bar Hill
    </h2>
    <img
      alt="track, asana"
      src=""
    />
  </li>
</div>
`;
